src/blocks/events/index.php

```diff
@@ -0,0 +1,188 @@
+		$align             = is_array( $attributes ) && isset( $attributes['align'] ) ? "align{$attributes['align']} " : '';
+
+		$class = 'wp-block-coblocks-events md:flex justify-between';
+		if ( isset( $attributes['className'] ) ) {
+			$class .= ' ' . $attributes['className'];
+		}
+
+		if ( isset( $attributes['align'] ) ) {
+			$class .= ' align' . $align;
+		}
+
+		$events_layout = sprintf( '<div class="%1$s">', esc_attr( $class ) );
+
+		foreach ( $events as $i => $event ) {
+			$page_num       = (int) ( $i / $attributes['eventsToShow'] );
+			$events_layout .= sprintf( '<div class="wp-block-coblocks-event-item" data-page="%1$s">', $page_num );
+
+			$dtstart           = $ical->ical_date_to_date_time( $event->dtstart_array[3] );
+			$dtend             = $ical->ical_date_to_date_time( $event->dtend_array[3] );
+			$start_date_string = strtotime( $dtstart->format( 'YmdHis' ) );
+			$end_date_string   = strtotime( $dtend->format( 'YmdHis' ) );
+			$day               = date( 'D', $start_date_string );
+			$month             = date( 'F', $start_date_string );
+			$day_of_month      = date( 'd', $start_date_string );
+			$start_time        = date( 'g:ia', $start_date_string );
+			$end_time          = date( 'g:ia', $end_date_string );
+			$time_string       = $start_time . ' - ' . $end_time;
+			$title             = $event->summary;
+			$desctiption       = $event->description;
+			$location          = $event->location;
+			$events_layout    .= sprintf(
+				'<div class="wp-block-coblocks-events__date"><p class="wp-block-coblocks-events__day display-block">%1$s</p><h4 class="wp-block-coblocks-events__month display-block">%2$s</h4><h4 class="wp-block-coblocks-events__year display-block">%3$s</h4></div>',
+				$day,
+				$month,
+				$day_of_month
+			);
+
+			$events_layout .= sprintf(
+				'<div class="wp-block-coblocks-events__content"><h4 class="wp-block-coblocks-events__title">%1$s</h4><p class="wp-block-coblocks-events__description">%2$s</p></div>',
+				$title,
+				$desctiption
+			);
+
+			$events_layout .= sprintf(
+				'<div class="wp-block-coblocks-events__details"><h5 class="wp-block-coblocks-events__time">%1$s</h5><p class="wp-block-coblocks-events__location">%2$s</p></div>',
+				$time_string,
+				$location
+			);
+
+			$events_layout .= '</div>';
+		}
+
+		if ( count( $events ) > $attributes['eventsToShow'] ) {
+			$events_layout .= sprintf(
+				'<div class="wp-block-coblocks-events__more-events-wrapper flex %1$s"><p>%1$s</p></div>',
+				__( 'More Events', 'coblocks' )
+			);
+		}
+
+		$events_layout .= '</div>';
+
+		return $events_layout;
+
+	} catch ( \Exception $e ) {
+
+		return '<div class="components-placeholder"><div class="notice notice-error">' . __( 'An error has occurred, check for calendar privileges, make sure it is public, or try again later.', 'coblocks' ) . '</div></div>';
```

Comment:
In the context of WordPress we should be using [`WP_Error`](https://developer.wordpress.org/reference/classes/wp_error/) for error reporting.